src/pipdeptree/_models/package.py

```diff
@@ -118,31 +110,69 @@ class DistPackage(Package):
 
     """
 
-    def __init__(self, obj: DistInfoDistribution, req: ReqPackage | None = None) -> None:
-        super().__init__(obj)
+    def __init__(self, obj: Distribution, req: ReqPackage | None = None) -> None:
+        super().__init__(obj.metadata["Name"])
+        self._obj = obj
         self.req = req
-        self._project_name = ""
 
     def requires(self) -> list[Requirement]:
-        return self._obj.requires()  # type: ignore[no-untyped-call,no-any-return]
+        req_list = []
+        req_name_list = []
+        if self._obj.requires:
+            for r in self._obj.requires:
+                req = Requirement(r)
+                is_extra_req = req.marker and contains_extra(str(req.marker))
+                if not is_extra_req and req.name not in req_name_list:
+                    req_list.append(req)
+                    req_name_list.append(req.name)
+        return req_list
 
     @property
-    def project_name(self) -> str:
-        if not self._project_name:
-            try:
-                self._project_name = metadata(self.key)["name"]
-            except (PackageNotFoundError, KeyError):
-                self._project_name = self._obj.project_name
-        return self._project_name
+    def editable(self) -> bool:
+        return self.direct_url_dict["editable"]
+
+    @property
+    def direct_url(self) -> str:
+        return self.direct_url_dict["url"]
+
+    @property
+    def raw_name(self) -> str:
+        return self.project_name
 
     @property
     def version(self) -> str:
         return self._obj.version  # type: ignore[no-any-return]
 
+    @property
+    def editable_project_location(self) -> str | None:
+        if self.direct_url:
+            from pip._internal.utils.urls import url_to_path  # noqa: PLC2701, PLC0415
+
+            return url_to_path(self.direct_url)
+
+        result = None
+        egg_link_path = egg_link_path_from_sys_path(self.raw_name)
+        if egg_link_path:
+            with Path(egg_link_path).open("r") as f:
+                result = f.readline().rstrip()
+        return result
+
+    @property
+    def direct_url_dict(self) -> dict[str, Any]:
+        result = {"editable": False, "url": None}
+
+        j_content = self._obj.read_text("direct_url.json")
+        if j_content:
+            drurl = DirectUrl.from_json(j_content)
+            result["url"] = drurl.url
+            result["editable"] = bool(drurl.is_local_editable)
```

Comment:
`is_local_editable` is actually just a function, so we can just do:
```suggestion
            result["editable"] = drurl.is_local_editable()
```

Reply:
Done.